src/pages/index.js

```diff
@@ -1,55 +1,160 @@
 import "./index.css";
 import Card from "../components/Card.js";
-import { cardList, config, userInfo } from "../utils/constants.js";
+import { config, userInfo } from "../utils/constants.js";
 import PopupWithImage from "../components/PopupWithImage.js";
 import PopupWithForm from "../components/PopupWithForm.js";
 import FormValidator from "../components/FormValidator.js";
 import Section from "../components/Section.js";
+import Api from "../components/Api.js";
 
-// What is a computer virus?
-// A terminal illness.
+let currentCard = undefined;
 
-// Popups
-const imagePopup = new PopupWithImage({
-  popupSelector: ".popup--open-image",
-});
-const userInfoPopup = new PopupWithForm(
-  { popupSelector: ".popup--edit-profile" },
-  (data) => userInfo.setUserInfo({ name: data.name, job: data.job })
-);
-const newPlacePopupButton = document.querySelector(".profile__add-button");
-const editInfoPopupButton = document.querySelector(
-  ".profile__edit-name-button"
+// Delete card Popup
+const deleteCardPopup = new PopupWithForm(
+  { popupSelector: ".popup--delete-card" },
+  () => {
+    deleteCardPopup.getForm().querySelector(".form__submit").textContent =
+      "Deleting...";
+    return api
+      .deleteCard({
+        cardId: currentCard.getCard().id,
+      })
+      .then(() => {
+        renderCards();
+        deleteCardPopup.close();
+      });
+  }
 );
+deleteCardPopup.setEventListeners();
 
-// Gallery
+// Create card
 function createCard(data) {
-  const card = new Card(data, ".card", () => {
-    imagePopup.open({ name: data.name, url: data.link });
+  const card = new Card(data, ".card", {
+    handleImageClick: () => {
+      imagePopup.open({ name: data.name, url: data.link });
+    },
+    deleteCallback: () => {
+      currentCard = card;
+      deleteCardPopup.getForm().querySelector(".form__submit").textContent =
+        "Delete";
+      deleteCardPopup.open();
+    },
+    likeCallback: () => {
+      return !card.isLiked
+        ? api.like({ cardId: card.getCard().id })
+        : api.unlike({ cardId: card.getCard().id });
+    },
   });
 
+  card.updateLike();
+
   return card.getCard().element;
 }
 
-const gallery = new Section(
-  {
-    items: cardList,
-    renderer: (data) => {
-      const cardElement = createCard(data);
-      gallery.addItem(cardElement);
-    },
-  },
-  ".gallery"
+// API
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  authToken: "7fef8830-65a5-4840-8327-1cd6adab3c4d",
+});
+
+// Render all initials information from API
+api.getUser().then((userData) => {
+  userInfo.setUserInfo({
+    name: userData.name,
+    job: userData.about,
+    id: userData._id,
+    pfpURL: userData.avatar,
+  });
+  userLoaded = true;
+});
+
+function renderCards() {
+  api.getCards().then((cards) => {
+    const gallery = new Section(
```

Comment:
To be able to reuse the `gallery` variable for creating a new card, you can define it outside the function with `let`, and then assign it inside `renderCards`

```js

let gallery;  // define

function renderCards() {
  api.getCards().then((cards) => {
    gallery = new Section(  // assign
      {
```


Then reuse `gallery` in `newPlacePopup` submit handler